Currently, trying to build the `master` branch leads to these errors:

```
root@811e4594c704:/docker-ros/ws# colcon build
Starting >>> zed_components
--- stderr: zed_components                               
 * Point Cloud Transport is available
/docker-ros/ws/src/target/zed_components/src/tools/src/sl_tools.cpp: In function ‘bool sl_tools::isZEDX(sl::MODEL)’:
/docker-ros/ws/src/target/zed_components/src/tools/src/sl_tools.cpp:677:30: error: ‘ZED_X_NANO’ is not a member of ‘sl::MODEL’
  677 |   if (camModel == sl::MODEL::ZED_X_NANO) {
      |                              ^~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/tools/src/sl_tools.cpp: In function ‘bool sl_tools::isZEDX(sl::MODEL)’:
/docker-ros/ws/src/target/zed_components/src/tools/src/sl_tools.cpp:677:30: error: ‘ZED_X_NANO’ is not a member of ‘sl::MODEL’
  677 |   if (camModel == sl::MODEL::ZED_X_NANO) {
      |                              ^~~~~~~~~~
In file included from /docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_bodytrk.cpp:15:
/docker-ros/ws/src/target/zed_components/src/zed_camera/include/zed_camera_component.hpp:563:7: error: ‘VoxelMeasureParameters’ in namespace ‘sl’ does not name a type
  563 |   sl::VoxelMeasureParameters mVoxelParams;
      |       ^~~~~~~~~~~~~~~~~~~~~~
In file included from /docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:15:
/docker-ros/ws/src/target/zed_components/src/zed_camera/include/zed_camera_component.hpp:563:7: error: ‘VoxelMeasureParameters’ in namespace ‘sl’ does not name a type
  563 |   sl::VoxelMeasureParameters mVoxelParams;
      |       ^~~~~~~~~~~~~~~~~~~~~~
In file included from /docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp:15:
/docker-ros/ws/src/target/zed_components/src/zed_camera/include/zed_camera_component.hpp:563:7: error: ‘VoxelMeasureParameters’ in namespace ‘sl’ does not name a type
  563 |   sl::VoxelMeasureParameters mVoxelParams;
      |       ^~~~~~~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp: In member function ‘void stereolabs::ZedCamera::getDepthParams()’:
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:672:30: error: ‘mVoxelParams’ was not declared in this scope
  672 |       double voxel_size_mm = mVoxelParams.voxel_size < 0 ?
      |                              ^~~~~~~~~~~~
In file included from /docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_objdet.cpp:15:
/docker-ros/ws/src/target/zed_components/src/zed_camera/include/zed_camera_component.hpp:563:7: error: ‘VoxelMeasureParameters’ in namespace ‘sl’ does not name a type
  563 |   sl::VoxelMeasureParameters mVoxelParams;
      |       ^~~~~~~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:685:44: error: ‘sl::VOXELIZATION_MODE’ has not been declared
  685 |         mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::FIXED;
      |                                            ^~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:687:44: error: ‘sl::VOXELIZATION_MODE’ has not been declared
  687 |         mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::LINEAR;
      |                                            ^~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:689:44: error: ‘sl::VOXELIZATION_MODE’ has not been declared
  689 |         mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::STEREO_UNCERTAINTY;
      |                                            ^~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp: In member function ‘void stereolabs::ZedCamera::fillCamInfo(const std::shared_ptr<sl::Camera>&, const sensor_msgs::msg::CameraInfo_<std::allocator<void> >::SharedPtr&, const sensor_msgs::msg::CameraInfo_<std::allocator<void> >::SharedPtr&, const std::string&, const std::string&, bool)’:
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:764:21: error: ‘ZED_X_NANO’ is not a member of ‘sl::MODEL’
  764 |     case sl::MODEL::ZED_X_NANO:  // RATIONAL_POLYNOMIAL
      |                     ^~~~~~~~~~
gmake[2]: *** [CMakeFiles/zed_camera_one_component.dir/build.make:76: CMakeFiles/zed_camera_one_component.dir/src/tools/src/sl_tools.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/docker-ros/ws/src/target/zed_components/src/zed_camera_one/src/zed_camera_one_component_main.cpp: In member function ‘bool stereolabs::ZedCameraOne::openZedCamera()’:
/docker-ros/ws/src/target/zed_components/src/zed_camera_one/src/zed_camera_one_component_main.cpp:983:40: error: ‘CAMERA_EXCEEDS_BANDWIDTH’ is not a member of ‘sl::ERROR_CODE’
  983 |     if (_connStatus == sl::ERROR_CODE::CAMERA_EXCEEDS_BANDWIDTH) {
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp: In member function ‘void stereolabs::ZedCamera::processPointCloud()’:
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:2677:24: error: ‘using std::__shared_ptr_access<sl::Camera, __gnu_cxx::_S_atomic, false, false>::element_type = class sl::Camera’ {aka ‘class sl::Camera’} has no member named ‘retrieveVoxelMeasure’; did you mean ‘retrieveMeasure’?
 2677 |         pc_err = mZed->retrieveVoxelMeasure(
      |                        ^~~~~~~~~~~~~~~~~~~~
      |                        retrieveMeasure
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:2678:58: error: ‘mVoxelParams’ was not declared in this scope
 2678 |           mMatCloud, sl::MEASURE::XYZBGRA, sl::MEM::CPU, mVoxelParams);
      |                                                          ^~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp: In member function ‘bool stereolabs::ZedCamera::handleDepthParams(const rclcpp::Parameter&, rcl_interfaces::msg::SetParametersResult&)’:
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3469:5: error: ‘mVoxelParams’ was not declared in this scope
 3469 |     mVoxelParams.voxel_size = voxel_size_mm <= 0 ?
      |     ^~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3483:5: error: ‘mVoxelParams’ was not declared in this scope
 3483 |     mVoxelParams.resolution_scale = static_cast<float>(param.as_double());
      |     ^~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp: In member function ‘void stereolabs::ZedCamera::getGeneralParams()’:
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp:1095:32: error: ‘ZED_X_NANO’ is not a member of ‘sl::MODEL’
 1095 |     mCamUserModel = sl::MODEL::ZED_X_NANO;
      |                                ^~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3497:7: error: ‘mVoxelParams’ was not declared in this scope
 3497 |       mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::FIXED;
      |       ^~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3497:42: error: ‘sl::VOXELIZATION_MODE’ has not been declared
 3497 |       mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::FIXED;
      |                                          ^~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3499:7: error: ‘mVoxelParams’ was not declared in this scope
 3499 |       mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::LINEAR;
      |       ^~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3499:42: error: ‘sl::VOXELIZATION_MODE’ has not been declared
 3499 |       mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::LINEAR;
      |                                          ^~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3501:7: error: ‘mVoxelParams’ was not declared in this scope
 3501 |       mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::STEREO_UNCERTAINTY;
      |       ^~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp:3501:42: error: ‘sl::VOXELIZATION_MODE’ has not been declared
 3501 |       mVoxelParams.resolution_mode = sl::VOXELIZATION_MODE::STEREO_UNCERTAINTY;
      |                                          ^~~~~~~~~~~~~~~~~
gmake[2]: *** [CMakeFiles/zed_camera_component.dir/build.make:76: CMakeFiles/zed_camera_component.dir/src/tools/src/sl_tools.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp: In member function ‘bool stereolabs::ZedCamera::startCamera()’:
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp:2909:40: error: ‘CAMERA_EXCEEDS_BANDWIDTH’ is not a member of ‘sl::ERROR_CODE’
 2909 |     if (mConnStatus == sl::ERROR_CODE::CAMERA_EXCEEDS_BANDWIDTH) {
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp:3088:42: error: ‘ZED_X_NANO’ is not a member of ‘sl::MODEL’
 3088 |   } else if (mCamRealModel == sl::MODEL::ZED_X_NANO) {
      |                                          ^~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp:3089:37: error: ‘ZED_X_NANO’ is not a member of ‘sl::MODEL’
 3089 |     if (mCamUserModel != sl::MODEL::ZED_X_NANO) {
      |                                     ^~~~~~~~~~
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp: In member function ‘void stereolabs::ZedCamera::publishCameraTFs(rclcpp::Time)’:
/docker-ros/ws/src/target/zed_components/src/zed_camera/src/zed_camera_component_main.cpp:5809:21: error: ‘ZED_X_NANO’ is not a member of ‘sl::MODEL’
 5809 |     case sl::MODEL::ZED_X_NANO:
      |                     ^~~~~~~~~~
gmake[2]: *** [CMakeFiles/zed_camera_component.dir/build.make:174: CMakeFiles/zed_camera_component.dir/src/zed_camera/src/zed_camera_component_bodytrk.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/zed_camera_component.dir/build.make:160: CMakeFiles/zed_camera_component.dir/src/zed_camera/src/zed_camera_component_objdet.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/zed_camera_component.dir/build.make:146: CMakeFiles/zed_camera_component.dir/src/zed_camera/src/zed_camera_component_video_depth.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/zed_camera_one_component.dir/build.make:132: CMakeFiles/zed_camera_one_component.dir/src/zed_camera_one/src/zed_camera_one_component_main.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:192: CMakeFiles/zed_camera_one_component.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake[2]: *** [CMakeFiles/zed_camera_component.dir/build.make:132: CMakeFiles/zed_camera_component.dir/src/zed_camera/src/zed_camera_component_main.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:166: CMakeFiles/zed_camera_component.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< zed_components [19.3s, exited with code 2]

Summary: 0 packages finished [19.4s]
  1 package failed: zed_components
  1 package had stderr output: zed_components
  3 packages not processed
```

--- a/zed_components/src/zed_camera/src/zed_camera_component_main.cpp
+++ b/zed_components/src/zed_camera/src/zed_camera_component_main.cpp
@@ -9608,7 +9608,7 @@ void ZedCamera::callback_setRoi(
       if (_nitrosDisabled) {
         if (mPubRoiMask.getTopic().empty()) {
           mPubRoiMask = image_transport::create_publisher(
-            this, mRoiMaskTopic, mQos.get_rmw_qos_profile());
+            this, mRoiMaskTopic, mQos.get_rmw_qos_profile(), mPubOpt);
           RCLCPP_INFO_STREAM(
             get_logger(), " * Advertised on topic: "
               << mPubRoiMask.getTopic());

--- a/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp
+++ b/zed_components/src/zed_camera/src/zed_camera_component_video_depth.cpp
@@ -217,7 +217,7 @@ void ZedCamera::initVideoDepthPublishers()
       ImageTopicType type = ImageTopicType::IMAGE) {
         ipcPub = create_ipc_pub(topic);
         set_transport_plugins(topic, type);
-        itPub = image_transport::create_publisher(this, topic, qos);
+        itPub = image_transport::create_publisher(this, topic, qos, mPubOpt);
         log_cam_pub(itPub);
       };
 
@@ -330,7 +330,7 @@ void ZedCamera::initVideoDepthPublishers()
   // Lambda to create and log CameraInfo publishers
   auto make_cam_info_pub = [&](const std::string & topic) {
       std::string info_topic = image_transport::getCameraInfoTopic(topic);
-      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, mQos);
+      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, mQos, mPubOpt);
       RCLCPP_INFO_STREAM(get_logger(), " * Advertised on topic: " << pub->get_topic_name());
       return pub;
     };
@@ -338,7 +338,7 @@ void ZedCamera::initVideoDepthPublishers()
   // Lambda to create and log CameraInfo publishers for image_transport or nitros
   auto make_cam_info_trans_pub = [&](const std::string & topic) {
       std::string info_topic = topic + "/camera_info";
-      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, mQos);
+      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, mQos, mPubOpt);
       RCLCPP_INFO_STREAM(get_logger(), " * Advertised on topic: " << pub->get_topic_name());
       return pub;
     };

--- a/zed_components/src/zed_camera_one/src/zed_camera_one_component_video.cpp
+++ b/zed_components/src/zed_camera_one/src/zed_camera_one_component_video.cpp
@@ -274,7 +274,7 @@ void ZedCameraOne::initVideoPublishers()
     ImageTopicType type = ImageTopicType::IMAGE) {
       ipcPub = create_ipc_pub(topic);
       set_transport_plugins(topic, type);
-      itPub = image_transport::create_publisher(this, topic, qos);
+      itPub = image_transport::create_publisher(this, topic, qos, _pubOpt);
       log_cam_pub(itPub);
     };
 
@@ -332,7 +332,7 @@ void ZedCameraOne::initVideoPublishers()
   // Lambda to create and log CameraInfo publishers
   auto make_cam_info_pub = [&](const std::string & topic) {
       std::string info_topic = image_transport::getCameraInfoTopic(topic);
-      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, _qos);
+      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, _qos, _pubOpt);
       RCLCPP_INFO_STREAM(get_logger(), "  * Advertised on topic: " << pub->get_topic_name());
       return pub;
     };
@@ -340,7 +340,7 @@ void ZedCameraOne::initVideoPublishers()
   // Lambda to create and log CameraInfo publishers for image_transport or nitros
   auto make_cam_info_trans_pub = [&](const std::string & topic) {
       std::string info_topic = topic + "/camera_info";
-      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, _qos);
+      auto pub = create_publisher<sensor_msgs::msg::CameraInfo>(info_topic, _qos, _pubOpt);
       RCLCPP_INFO_STREAM(
         get_logger(),
         "  * Advertised on topic: " << pub->get_topic_name());

